src/Microsoft.AspNetCore.Mvc.Modules/Extensions/ServiceCollectionBuilderExtensions.cs

```diff
@@ -67,6 +69,7 @@ public static IMvcCoreBuilder AddExtensionsApplicationParts(this IMvcCoreBuilder
             var availableExtensions = extensionManager.GetExtensions();
             using (logger.BeginScope("Loading extensions"))
             {
+                ConcurrentBag<Assembly> bagOfAssemblies = new ConcurrentBag<Assembly>();
                 Parallel.ForEach(availableExtensions, new ParallelOptions { MaxDegreeOfParallelism = 4 }, ae =>
```

Comment:
Are you saying `GetExtensions()` is not actually loading them, but only when we enumerate its result?